src/cryptography/hazmat/backends/openssl/backend.py

```diff
@@ -136,6 +138,39 @@ def _encode_basic_constraints(backend, basic_constraints):
     return pp, r
 
 
+def _encode_subject_alt_name(backend, san):
+    general_names = backend._lib.GENERAL_NAMES_new()
+    assert general_names != backend._ffi.NULL
+    general_names = backend._ffi.gc(
+        general_names, backend._lib.GENERAL_NAMES_free
+    )
+
+    for alt_name in san:
+        gn = backend._lib.GENERAL_NAME_new()
+        assert gn != backend._ffi.NULL
+        if isinstance(alt_name, x509.DNSName):
+            gn.type = backend._lib.GEN_DNS
+            ia5 = backend._lib.ASN1_IA5STRING_new()
+            assert ia5 != backend._ffi.NULL
+            value = idna.encode(alt_name.value)
```

Comment:
Handle wildcards here or in a followup PR? Leading periods as well, but that's not hugely important until we support NameConstraints construction I suppose.